### ISSUE TITLE
Bug Report: Multiple Arbitrary File Deletion vulnerabilities

### ISSUE THREAD

**Issue by hibiki-sama** [kanagawa-okinamiura/KANAcms#1]
Vulnerability Name: Multiple Arbitrary File Deletion

Date of Discovery: 30 August 2021

Product version: 2.5.7 [Download link](https://github.com/kanagawa-okinamiura/KANAcms/releases/download/KANA/kanacms.rar)

Author: hibiki-sama

Vulnerability Description: When unsanitized user input is supplied to a file deletion function, an arbitrary file deletion vulnerability arises. This occurs in PHP when the unlink() function is called and user input might affect portions of or the whole affected parameter, which represents the path of the file to remove, without sufficient sanitization. Exploiting the vulnerability allows an attacker to delete any file in the web root (along with any other file on the server that the PHP process user has the proper permissions to delete). Furthermore, an attacker can leverage the capability of arbitrary file deletion to circumvent certain webserver security mechanisms such as deleting .htaccess file that would deactivate those security constraints.

Proof of Concept 1

Vulnerable URL: /user/ppsave.php
Vulnerable Code: line 68 - kanacms\user

![1](https://user-images.githubusercontent.com/65580366/131305955-5d89dcea-1b04-42b5-87c9-a8d9e0016473.png)

It can be found that there is no verification, just judge whether it is the same as the previous or default, and then use unlink to delete the file as long as the file exists

Therefore, the vulnerability analysis and utilization are very simple

![22](https://user-images.githubusercontent.com/65580366/131306083-351a9414-1c37-429e-9bdc-291bc8f7e12a.png)

We deleted the installed lock file / install / install.lock

![t01cdfee4b0f1e15e9f](https://user-images.githubusercontent.com/65580366/131306228-9a0477bc-9422-4a56-8693-ad58a956ed20.png)

Proof of Concept 2

Vulnerable URL: /user/adv.php
Vulnerable Code: line 77 - kanacms\user\/adv.php

There is an arbitrary file deletion vulnerability. I have to say that the system is really problematic in judging this

The file causing the problem is in / user / adv.php

It is also a problem caused by the comparison between oldimg and img

![33](https://user-images.githubusercontent.com/65580366/131306791-ff5d82f2-4e45-455d-97e8-ea5688056105.png)

Similar to the above analysis, it only judges whether it is the same as the original, and then splices.. / and directly calls unlink, so the use is also very simple

Just delete the hidden of the form attribute in HTML, and then directly enter the file name you want to delete

**Comment by kanagawa-okinamiura** [comment]
Yes, according to our adjustment, these vulnerabilities do exist. Please wait a little while for us to modify them. Thank you

**Comment by hibiki-sama** [comment]
In fact, it's best to add some filtering means to prevent these from happening again. Thank yo
